### ISSUE TITLE
Include shape information of output in the gRPC PredictionResponse 

### ISSUE THREAD

**Issue by IamMohitM** [pytorch/serve#1683]
### 🚀 The feature

Include the shape of the predicted output in the PredictionResponse message in grpc. For example, if a segmentation model is served, which returns a segmented output of the input shape. Include the shape of the segmented output (in this case, same as input shape) in the PredictedResponse.

### Motivation, pitch

Currently, the PredictionResponse only contains a byte string of the model output. This is fine in cases where the responses are classification responses. However, certain models could have non-deterministic output shape which may not be easily derived from the input shapes. 

For example, the output shape might be (17, 75, 64) for an image of size (720, 720) and (17, 100, 64) for an image of size (800, 800)

Currently, there are two ways to convert the bytes to required tensor - shape

- Store the shape of the output array and use this value to reshape. Not sophisticated if the input shape keeps changing
- Use `ast.literal_eval` on the decoded string

It can't be done 

### Alternatives

Perhaps we can return <map, string> in the PredictionResponse to make it more customisable.

**Comment by maaquib** [comment]
@IamMohitM You can write a [custom handler](https://github.com/pytorch/serve/blob/master/docs/custom_service.md#custom-handlers) in this case, which would augment the response with the shape

**Comment by IamMohitM** [comment]
Thanks @maaquib. That does make sense. I have tried adding that to my custom handler:

```
def postprocess(self, data):
     return [data, data.shape]
```

However, the problem happens at the client. Apart from literal eval, I can't efficiently seem to decode the bytes I receive from torch serve. This is related to another issue I created - #1684. Is there any other way to decode the bytes that I receive apart from `ast.literal_eval`?
